job-dsls/jobs/reduced_drools_release.groovy

```diff
@@ -0,0 +1,241 @@
+
+def redRelease='''
+pipeline {
+    agent {
+        label 'kie-rhel7 && kie-mem24g'
+    }
+    tools {
+        maven 'kie-maven-3.6.3'
+        jdk 'kie-jdk1.8'
+    }
+    stages {
+        stage('CleanWorkspace') {
+            steps {
+                cleanWs()
+            }
+        }
+        stage('Checkout droolsjbpm-build-bootstrap') {
+            steps {
+                checkout([$class: 'GitSCM', branches: [[name: '$baseBranch']], browser: [$class: 'GithubWeb', repoUrl: 'git@github.com:$organization/droolsjbpm-build-bootstrap.git'], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'RelativeTargetDirectory', relativeTargetDir: 'droolsjbpm-build-bootstrap']], submoduleCfg: [], userRemoteConfigs: [[credentialsId: 'kie-ci-user-key', url: 'git@github.com:$organization/droolsjbpm-build-bootstrap.git']]])
+                dir("${WORKSPACE}" + '/droolsjbpm-build-bootstrap') {
+                    sh  'pwd \\n' +
+                        'git branch \\n' +
+                        'git checkout -b $baseBranch'
+                }
+            }    
+        }
+        stage ('Clone others') {
+            steps {
+                sshagent(['kie-ci-user-key']) {
+                    sh 'sh droolsjbpm-build-bootstrap/script/release/01_cloneBranches.sh $baseBranch'
+                }    
+            }
+        }
+        // checks if release branch already exists
+        stage ('Check if branch exists') {
+            steps{
+                sshagent(['kie-ci-user-key']) {
+                    dir("${WORKSPACE}" + '/droolsjbpm-build-bootstrap') {
+                        script {
+                            branchExists = sh(script: 'git ls-remote --heads origin ${releaseBranch} | wc -l', returnStdout: true).trim()
+                        }
+                    }
+                }
+            }
+        }
+        stage ('log results') {
+            steps {
+                echo 'branchExists: ' + "$branchExists"
+                script {
+                    if ( "$branchExists" == "1") {
+                        echo "branch exists"
+                    } else {
+                        echo "branch does not exist"
+                    }
+                }
+            }
+        }
+        // if release branches doesn't exist they will be created
+        stage('Create release branches') {
+            when{
+                expression { branchExists == '0'}
+            }
+            steps {
+                sshagent(['kie-ci-user-key']) {
+                    sh 'sh droolsjbpm-build-bootstrap/script/release/02_createReleaseBranches.sh $releaseBranch'
+                }
+            }
+        }
+        // part of the Maven rep will be erased
+        stage ('Remove M2') {
+            steps {
+                sh "sh droolsjbpm-build-bootstrap/script/release/eraseM2.sh $m2Dir"
+            }
+        }
+        stage ('replace repository-list.txt and change branched-7-repository-list.txt') {
```

Comment:
Shouldn't this stage be more up, before you clone the other repositories?

Reply:
right now it is in a later step - but you're right, will change this